events/2023-google-summer-of-code/index.markdown

```diff
@@ -94,6 +76,37 @@ For general MoveIt GSoC questions you can contact [Mark Moll](mailto:mark@pickni
 * Preferred skills: MoveIt 2, ROS 2, C++, transformations or geometry for robotics
 * Mentor: [Sebastian Castro](mailto:sebastian.castro@picknik.ai), [Tyler Weaver](mailto:tyler@picknik.ai)
 
+### Update OMPL and ompl_interface [MoveIt 2]
+
+* The Open Motion Planning Library (OMPL) provides the default planning pipeline for MoveIt. It is a library that contains many sampling-based planning algorithms (like PRM and RRT), but is starting to show its age (development started in 2010!). This project is focused on modernizing OMPL, adopting more idiomatic C++17 patterns and avoiding things like using raw pointers and type casting. There are also many OMPL features that are not well supported within MoveIt's `ompl_interface` just yet. Finally, there are features in the `ompl_interface` that seem generally useful and could be abstracted out of this.
+* Possible tasks (you are not required to do all of this; your application should make clear which parts you want to focus on):
+  * Use `clang-tidy` to identify areas of improvement in OMPL. Remove raw pointer usage in the most commonly used parts. Eliminate ambiguity in the API about the lifetime of objects.
+  * Make constrained motion planning work more generally. See, e.g., this [work-in-progress PR](https://github.com/ros-planning/moveit2/pull/1946) for a specific (but very important) use case.
+  * Generalize things in `ompl_interface` and move them to `moveit_core`.
+* Project size: Large (350 hours)
+* Difficulty: hard
+* Preferred skills: MoveIt 2, OMPL, C++, good understanding of algorithms
+* Mentor: [Zak Kingston](mailto:zak@rice.edu), [Mark Moll](mailto:mark@picknik.ai), [Henning Kayser](mailto:henningkayser@picknik.ai)
+
+### Bullet collision checking integration [MoveIt 1/2]
+
+* Bullet support is partially done, but more work is needed. [A memory leak issue](https://github.com/ros-planning/moveit_task_constructor/issues/232#issuecomment-774249206) was identified that needs to be fixed. Support for continuous collision checking would be nice. Related pull requests include [PR 2897](https://github.com/ros-planning/moveit/pull/2897) and [PR 2838](https://github.com/ros-planning/moveit/pull/2838).
+* Project size: medium (175 hours)
+* Expected outcome: Continuous collision checking is supported and the memory leak is eliminated.
+* Difficulty: hard
+* Preferred skills: MoveIt, Collision Checking, and C++
+* Mentor: [Simon Schmeisser](mailto:simon.schmeisser@optonic.com), [Michael Görner](mailto:goerner@informatik.uni-hamburg.de)
+
+### Improved benchmarking support [MoveIt 1]
```

Comment:
Why is this MoveIt 1 specific? Could this project include porting to MoveIt 2 as part of its tasks?

Reply:
It doesn't *have* to be, but all the preliminary work mentioned in the description is MoveIt 1-based. @v4hn @simonschmeisser do you see a way that this can be a successful MoveIt 2-based project?

Reply:
The main task of this project is meant to *build content* for the benchmark suite and only modify the suite itself where necessary to get reasonable benchmarks implemented. I'm afraid that adding the dumb, but labor intense task of porting to ROS2 will just entail that we still don't have any content or CI setup by the end of the project.

The current implementation is based around ROS1 launch files and ROS parameters in core parts, so porting will incur quite some overhead and will likely raise a lot of questions about the basic design again. If some maintainer who worked on ROS2 parameter handling before and is interested in sane design (@tylerjw?)  is willing to supervise such an effort we might add it as a project and I can co-supervise it, but I am way more interested in the actual benchmark design/CI integration.

Note that a port to ROS2 could be a mostly independent work on the side as well or a clear follow-up, but I do not consider it a goal of this project.

Reply:
This is where I admit I'm questioning how much I want to be involved in mentoring students at all. This is not because it is not valuable work; it is because there is only so much time, and there are other things I'd rather do.